### PR TITLE
make symbolic map lemmas available for haskell backend as well

### PR DESCRIPTION
**Pull request by daejunpark**
Only the last commit contains meaningful changes. Others are reorg changes.

### CODE REVIEW

File: k-distribution/tests/regression-new/map-symbolic-tests/Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 DEF=map-tests
 EXT=mtest
 TESTDIR=.
-KOMPILE_BACKEND=java
+KOMPILE_BACKEND=haskell
 KPROVE_FLAGS=--restore-original-names --no-alpha-renaming --warnings none
 
 include ../../../include/kframework/ktest.mak
```
Comment (ehildenb) [inline k-distribution/tests/regression-new/map-symbolic-tests/Makefile]: We should probably run these tests with both backends for now.

Comment (daejunpark) [reply k-distribution/tests/regression-new/map-symbolic-tests/Makefile]: Yep, I know. I just wanted to test quickly, but doesn't seem to be a good idea, making Jenkins too loaded. Will not do that later.

Comment (ehildenb) [reply k-distribution/tests/regression-new/map-symbolic-tests/Makefile]: Well you can also test locally instead. Just run `make` in this directory after building.

Comment (daejunpark) [reply k-distribution/tests/regression-new/map-symbolic-tests/Makefile]: Yeah, I thought that K Jenkins build is quicker than compiling the haskell backend in my machine, which was wrong.

**Comment by daejunpark** [comment]
I think we need to make this test similar to the KEVM lemmas-spec.k.  The haskell backend output contains too verbose.

**Comment by daejunpark** [comment]
Hmm. `in_keys` doesn't seem to work as expected in the haskell backend..

**Comment by daejunpark** [comment]
OK. @ssaha6 will transform this test to make it similar to the lemmas-spec.k test.

**Pull request closed without merging**